### ISSUE TITLE
Multi-pass fix: N-pass oscillation cycles are not detected

### ISSUE THREAD

**Issue by saphryn** [markuplint/markuplint#3891]
## Bug Description

The multi-pass fix loop in `MLCore` detects 2-pass exact oscillation (A → B → A) but misses longer cycles. A fix sequence A → B → C → A will loop until MAX_FIX_PASSES (10) is reached, then silently stop with some fixes unapplied and no indication of which ones.

## Root Cause

In `packages/@markuplint/ml-core/src/ml-core.ts`, the cycle detection logic:

```typescript
// Only catches 2-pass oscillation
if (previousCode !== undefined && result.output === previousCode) {
  log('fix pass %d: cycle detected (output matches pass %d), stopping', pass, pass - 2);
  currentCode = result.output;
  break;
}
```

This only compares the current output against `previousCode` (one pass back). A 3-element cycle (A → B → C → A) produces output C in pass 2, which does not match previousCode (= B), so detection fails.

## Impact

- **Silent fix failure**: Fixes are applied up to MAX_FIX_PASSES and then stopped. The `reachedMaxPasses: true` flag in `FixSummary` tells you the max was hit, but gives no information about which fixes failed to apply or why.
- **No user signal**: The caller receives `fixedCode` with no indication that not all fixes were applied.
- **Incorrect markup**: Some violations are fixed but others silently remain, giving a false impression of correctness.

## Reproduction

1. Create markup where fix A triggers a new violation in pass 2, fix B resolves it in pass 3, but fix C reintroduces the original problem in pass 4 — a 3-step oscillation.
2. Call `verify({ fix: true })`
3. Observe: loop runs 10 times, `reachedMaxPasses: true`, but no indication which fixes were skipped.

## Proposed Fix

Track a history buffer of the last N outputs (where N = MAX_FIX_PASSES). On each pass, check whether the current output matches any entry in the history, not just the immediately previous one. If a match is found, report the oscillation cycle explicitly:

```typescript
const outputHistory = new Set<string>();
// In the loop:
if (outputHistory.has(result.output)) {
  // Cycle detected — report which fixes oscillate
  break;
}
outputHistory.add(result.output);
```

Additionally, surface the list of fixes that could not be applied due to oscillation in the `FixSummary`.

## Additional Context

This affects the TypeScript core. The Fix phase (Phase 3-3~3-6) in the Rust v6 rewrite (https://github.com/markuplint/markuplint/issues/3178) should address this with awareness of the TypeScript behavior.

Affected file: `packages/@markuplint/ml-core/src/ml-core.ts` (`#multiPassFix` method)

**Comment by YusukeHirao** [comment]
Confirmed: the cycle detection at `#multiPassFix` only compares against the output from one pass back, so an A → B → C → A cycle runs until `MAX_FIX_PASSES` and stops with only `reachedMaxPasses: true` as a signal.

**Decision: fix in v5 RC**, in the same PR as #3890 (both changes are in `#multiPassFix`):

- Keep a history of pass outputs and detect when the current output matches any previous pass, not just the immediately preceding one.
- Log the detected cycle length via the existing debug logger.

Note: no real-world 3-cycle has been observed so far — this is a theoretical gap — but the fix is small and the detection cost (at most `MAX_FIX_PASSES` strings) is negligible.
